homeassistant/components/hyundai_kia_connect/binary_sensor.py

```diff
@@ -0,0 +1,206 @@
+        key="rear_right_seat_heater_is_on",
+        name="Rear Right Seat Heater",
+        device_class=BinarySensorDeviceClass.HEAT,
+        on_icon="mdi:car-seat-heater",
+        off_icon="mdi:car-seat-heater",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="is_locked",
+        name="Lock",
+        device_class=BinarySensorDeviceClass.LOCK,
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="front_left_door_is_open",
+        name="Front Left Door",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-door",
+        off_icon="mdi:car-door",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="front_right_door_is_open",
+        name="Front Right Door",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-door",
+        off_icon="mdi:car-door",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="back_left_door_is_open",
+        name="Back Left Door",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-door",
+        off_icon="mdi:car-door",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="back_right_door_is_open",
+        name="Back Right Door",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-door",
+        off_icon="mdi:car-door",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="trunk_is_open",
+        name="Trunk",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-back",
+        off_icon="mdi:car-back",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="hood_is_open",
+        name="Hood",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car",
+        off_icon="mdi:car",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="ev_battery_is_charging",
+        name="EV Battery Charge",
+        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="ev_battery_is_plugged_in",
+        name="EV Battery Plug",
+        device_class=BinarySensorDeviceClass.PLUG,
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="fuel_level_is_low",
+        name="Fuel Low Level",
+        device_class=BinarySensorDeviceClass.LIGHT,
+        on_icon="mdi:gas-station-off",
+        off_icon="mdi:gas-station",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="data",
+        name="Debug Data",
+    ),
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up binary_sensor platform."""
+    coordinator = hass.data[DOMAIN][config_entry.unique_id]
+    entities = []
+    for vehicle_id in coordinator.data.vehicles.keys():
+        vehicle: Vehicle = coordinator.data.vehicles[vehicle_id]
+        for description in SENSOR_DESCRIPTIONS:
+            if getattr(vehicle, description.key, None) is not None:
```

Comment:
What does `None` mean in this context?
Currently not available? Not available ever?
Could it be `None` because of a temporary communication error?

Reply:
None means sensor is not viable for this device.
as an example, EV sensors are not available in combustion engine vehicles or low fuel light is missing from EV cars.
If these data points are missing, they are none.